### PR TITLE
Performance/logging/exception handling improvements

### PR DESCRIPTION
**Pull request by benj58xu**
This contains multiple improvements.
1. Convert to use the common RestClient, instead of its own implementation
2. Performance improvement by using indexed field in CollectorItem, instead of description
3. Capture exceptions so that errors of one dashboard will not stop continued processing of remaining dashboards
4. A number of logging improvements

### CODE REVIEW

File: src/test/java/com/capitalone/dashboard/collector/AuditCollectorTaskTest.java [hunk 1]
```diff
@@ -2,14 +2,7 @@
...
 import com.capitalone.dashboard.collector.config.TestConfig;
-import com.capitalone.dashboard.model.Dashboard;
-import com.capitalone.dashboard.model.AuditResult;
-import com.capitalone.dashboard.model.AuditType;
-import com.capitalone.dashboard.model.Audit;
-import com.capitalone.dashboard.model.AuditStatus;
-import com.capitalone.dashboard.model.DataStatus;
-import com.capitalone.dashboard.model.DashboardType;
-import com.capitalone.dashboard.model.CollectorItem;
+import com.capitalone.dashboard.model.*;
```
Comment (nameisaravind) [inline src/test/java/com/capitalone/dashboard/collector/AuditCollectorTaskTest.java:5]: @benj58xu can import only needed objects?

Comment (benj58xu) [reply src/test/java/com/capitalone/dashboard/collector/AuditCollectorTaskTest.java:5]: Wonder if there is an intellij setting? It does this automatically.

Comment (benj58xu) [reply src/test/java/com/capitalone/dashboard/collector/AuditCollectorTaskTest.java:5]: And I am also curious why people here insist a different style from others?

Comment (benj58xu) [reply src/test/java/com/capitalone/dashboard/collector/AuditCollectorTaskTest.java]: Changed.

File: src/main/java/com/capitalone/dashboard/collector/AuditCollectorUtil.java [hunk 2]
```diff
@@ -797,6 +804,8 @@ protected CollectorItem createCollectorItem(AuditType auditType){
             collectorItem.setLastUpdated(System.currentTimeMillis());
             collectorItem.setDescription(description);
             collectorItem.getOptions().put(OPT_DBRD_ID, getDashboard().getId());
+            collectorItem.getOptions().put(OPT_URL, description);
```
Comment (nameisaravind) [inline src/main/java/com/capitalone/dashboard/collector/AuditCollectorUtil.java:807]: texted in slack about this line..

**Review by nameisaravind — APPROVED** [review]
lgtm

**Review by rvema — APPROVED** [review]
lgtm

**Pull request merged**